redcap_bridge/server_elab_interface.py

```diff
@@ -1,18 +1,55 @@
 import json
 import os
 import elabapi_python
+import pandas as pd
 
 from redcap_bridge.utils import conversion_csv_to_json
 
 
-def download_experiment(server_config_json, experiment_id):
+def download_experiment(server_config_json, experiment_id, experiment_axis):
+    """
+    Download an individual experiment.
+
+    Parameters
+    ----------
+    server_config_json: str
+        Path to the json file containing the redcap url, api token and required external modules
+    experiment_id: int
+        ID of the experiment you want to download
+    experiment_axis: str
+        Option to control whether experiments are arranged in columns or rows
+
+    Returns
+    -------
+        status_code: Return code of the API request
+        df: Dataframe of the experiment
+    """
+
     api_client = get_elab_config(server_config_json)
     experiment_api = elabapi_python.ExperimentsApi(api_client)
 
     exp = experiment_api.get_experiment_with_http_info(experiment_id)
     status_code = exp[1]
 
-    return exp, status_code
+    metadata = json.loads(exp[0].metadata)
+
+    extra_fields_data = metadata.get("extra_fields", {})
+    unwanted_columns = ["position", "options", "allow_multi_values", "blank_value_on_duplicate"]
+
+    print(f'{extra_fields_data}')
+
+    if experiment_axis == "columns":
+        df = pd.DataFrame.from_dict(extra_fields_data, orient='columns')
+        df = df.drop(unwanted_columns, axis=0)  # Delete unwanted columns
+    elif experiment_axis == "row":
+        df = pd.DataFrame.from_dict(extra_fields_data, orient='index')
+        df = df.drop(unwanted_columns, axis=1)
+    else:
+        raise ValueError("Invalid experiment_axis value. Must be 'columns' or 'row'.")
```

Comment:
To be more consistent you should either use plural or singular for both, so ('columns' & 'rows') or ('column' and 'row')

Reply:
yes ok, i'll modify that